server/service/schedule/schedule.go

```diff
@@ -0,0 +1,219 @@
+package schedule
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/getsentry/sentry-go"
+	kitlog "github.com/go-kit/kit/log"
+	"github.com/go-kit/log/level"
+)
+
+type schedule struct {
+	ctx        context.Context
+	name       string
+	instanceID string
+	Logger     kitlog.Logger
+
+	muChecks       sync.Mutex
+	configInterval time.Duration
+	schedInterval  time.Duration
+	locker         Locker
+	configCheck    func(start time.Time, wait time.Duration) (*time.Duration, error)
+	preflightCheck func() bool
```

Comment:
`preflightCheck` is meant for uses cases like vulnerabilities where we want to check something before each run of the schedule (for example, confirming `CurrentInstanceChecks`). The preflight checks run on same interval as the schedule whereas the config check runs on its own interval.